src/json/json.cc

```diff
@@ -258,6 +258,130 @@ STATIC JsonUtilCode parseSetCmdArgs(ValkeyModuleString **argv, const int argc, S
     return JSONUTIL_SUCCESS;
 }
 
+typedef struct {
+    ValkeyModuleString *key_str;    // required
+    ValkeyModuleKey *key;
+    const char *path;               // required
+    const char *json;               // required
+    size_t json_len;
+    bool is_root_path;
+    size_t doc_size;
+} MSetCmdArgs;
+
+typedef struct {
+    size_t num_keys;
+    MSetCmdArgs *args_list;
+} MSetCmdArgsList;
+
+void FreeMSetCmdArgs(ValkeyModuleCtx *ctx, MSetCmdArgs *args_list, size_t num_keys) {
```

Comment:
The entire method is problematic that causes memory corruption. We can discuss at the next meeting.